### PR TITLE
rollouts: add package cluster matcher

### PR DESCRIPTION
**Pull request by ChristopherFry**
This pull request adds package cluster matching to the rollouts proof of concept.

### CODE REVIEW

File: rollouts/pkg/packageclustermatcher/packageclustermatcher.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	}
+
+	return allClusterPackages, nil
+}
+
+func isPackageClusterMatch(expr string, cluster, rolloutPackage map[string]interface{}) (bool, error) {
+	env, err := cel.NewEnv(
+		cel.Declarations(
+			decls.NewVar("cluster", decls.Dyn),
+			decls.NewVar("rolloutPackage", decls.Dyn),
```
Comment (droot) [inline rollouts/pkg/packageclustermatcher/packageclustermatcher.go:91]: should we rename `rolloutPackage` to just `package` ?

File: rollouts/pkg/packageclustermatcher/packageclustermatcher.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	}
+}
+
+func (m *PackageClusterMatcher) GetClusterPackages(matcher gitopsv1alpha1.PackageToClusterMatcher) ([]ClusterPackages, error) {
+	clusters := m.clusters
+	packages := m.packages
+
+	allClusterPackages := []ClusterPackages{}
+
+	for _, cluster := range clusters {
```
Comment (droot) [inline rollouts/pkg/packageclustermatcher/packageclustermatcher.go:51]: nit: I think we can constructs list of celCluster and list of celPackages separately and then loop over to run matchers. We can avoid costs of constructing `celPackages` for each clusters.

**Review by droot — APPROVED** [review]
Looks good. Minor comment.

File: rollouts/pkg/packageclustermatcher/packageclustermatcher.go (hunk 1 above)
Comment (ChristopherFry) [reply rollouts/pkg/packageclustermatcher/packageclustermatcher.go:91]: Unfortunately `package` is a resolved word in [CEL](https://github.com/google/cel-spec/blob/master/doc/langdef.md#syntax). I'm up for other suggestions though. `rolloutPackage` is a little wordy.

**Pull request merged**